--- user.rb	2017-10-16 20:04:08.000000000 +0200
+++ user_patched.rb	2017-10-21 11:02:14.821835100 +0200
@@ -109,6 +109,8 @@
 
       def find_ldap_person(auth_hash, adapter)
         by_uid = Gitlab::LDAP::Person.find_by_uid(auth_hash.uid, adapter)
+        # The `uid` might actually be a UUID. Try it next.
+        by_uid = Gitlab::LDAP::Person.find_by_uuid(auth_hash.uid, adapter)
         # The `uid` might actually be a DN. Try it next.
         by_uid || Gitlab::LDAP::Person.find_by_dn(auth_hash.uid, adapter)
       end
